DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/MVP.Script.txt

```diff
@@ -0,0 +1,327 @@
+	<label id="score" pos="0 -6" z-index="0" size="80 8" text="# Points" textfont="GameFontRegular" textsize="4" halign="center" valign="center"/>
+	<label id="description" pos="0 -12" z-index="0" size="80 16" text="" textfont="GameFontRegular" textsize="2" halign="center" valign="center2"/>
+	</framemodel>
+	
+	<frame id="mvp-frame" z-index="-1" hidden="1">
+	<frame id="title-frame" pos="0 40" halign="center" valign="center">	
+	<label id="title" z-index="0" size="150 12" text="M$n $mV$n $mP" textfont="GameFontExtraBold" textsize="22" halign="center" valign="center2"/>			
+	</frame>
+	<quad id="car1" pos="0 -22" z-index="0" size="55 42" bgcolor="0000" halign="center" valign="center" scriptevents="1"/>
+	<quad id="car2" pos="-61 -9.5" z-index="0" size="55 37" bgcolor="0000" halign="center" valign="center" scriptevents="1"/>
+	<quad id="car3" pos="61 -9.5" z-index="0" size="55 37" bgcolor="0000" halign="center" valign="center" scriptevents="1"/>
+	
+	<frameinstance id="mvp0" pos="0 -40" modelid="player" hidden="1" />
+	<frameinstance id="mvp1" pos="-65 -25" modelid="player" hidden="1" />
+	<frameinstance id="mvp2" pos="65 -25" modelid="player" hidden="1" />
+	<!-- <quad pos="0 0" z-index="-2" size="320 200" image="file://Media/Manialinks/Nadeo/TMNext/Menus/Background_marketing.dds" opacity="1" halign="center" valign="center" colorize="0af" /> -->
+	<quad id="background" pos="0 0" z-index="-2" size="320 200" image="file:///Media/Manialinks/Nadeo/TMNext/Menus/PageGarage/Garage_background.png" opacity="1" halign="center" valign="center" />
+	<frame size="360 80" valign="center" halign="center" pos="0 -10">
+	<camera id="camera" pos="0 10" size="360 180" valign="center" halign="center" />
+	</frame>
+	</frame>
+	
+	<script><!--
+	#Include "AnimLib" as AL
+	
+	#Struct K_Item {
+		Ident ItemId;
+		Vec3 Position;
+	}
+	
+	declare Ident SceneId;
+	declare K_Item[] Items;
+	
+	{{{ UIShared::GetTeamColorNetreadFunctions() }}}
+	{{{ UIShared::GetUtilFunctions() }}}
+	
+	/** 
+	* Gets CSmScore from array by index number
+	*/
+	CSmScore GetScore(Integer[CSmScore] Array, Integer Index) {
+		declare Integer x = 0;
+		foreach(Key => Value in Array) {
+			if (x == Index) {
+				return Key;
+			}
+			x += 1;
+		}
+		return Null;
+	}
+	
+	/**
+	* Get matchpoints, order is from z to a
+	*/
+	Integer[CSmScore] GetMatchPoints() {
+		declare Integer[CSmScore] Output;
+		foreach(Score in Scores) {
+			declare netread Integer Net_FlagRush_MatchPoints for Score;
+			//if (Net_FlagRush_MatchPoints > 0) 
+			Output[Score] = Net_FlagRush_MatchPoints;			
+			
+		}
+		return Output.sortreverse();
+	}
+	
+	/**
+	* Get Best player of FlagsScored
+	*/
+	CSmScore GetFlagsScored() {
+		declare Integer[CSmScore] Output;
+		foreach(Score in Scores) {
+			declare netread Integer Net_FlagRush_FlagsScored_Match for Score;
+			Output[Score] = Net_FlagRush_FlagsScored_Match;			
+		}
+		Output = Output.sortreverse();
+		return GetScore(Output, 0);
+	}
```

Comment:
Functionname doesn't really describe what it's actually doing. Also, if you want to get the score which has the most flags scored, you could just:
```suggestion
	/**
	* Get Best player of FlagsScored
	*/
	CSmScore GetScoreMostFlagsScored() {
		declare CSmScore BestScore;
		declare Integer MaxFlagScored = 0;
		foreach (Score in Scores) {
			declare netread Integer Net_FlagRush_FlagsScored_Match for Score;
			if (Net_FlagRush_FlagsScored_Match > MaxFlagsScored) {
				BestScore = Score;
				MaxFlagsScores = Net_FlagRush_FlagsScored;
			}
		}
		return BestScore;
	}
```

Not weird reverse sorting or using that GetScore function.